report-ng/app/src/components/classes/classes.ts

```diff
@@ -313,103 +311,110 @@ export class Classes extends AbstractViewModel {
-            }
+    private _removeFilter(filterType: FilterType, filterObject?){
 
-            // insert classes that are not selected as filter anymore back into filter class select options
-            if(chipType == ChipType.CLASS && !this._filteredClassStatistics.some(stat => stat.classIdentifier == filterObject)){
-                this._filteredClassStatistics.push(this._executionStatistics.classStatistics.find(stat => stat.classIdentifier == filterObject));
-                this._filteredClassStatistics.sort((a,b) => a.classIdentifier.localeCompare(b.classIdentifier))
-            }
+        var filterTypeArray = this.queryParams[filterType].split("~");
+        const index = filterTypeArray.indexOf(filterObject);
+        filterTypeArray.splice(index, 1);
+        if(filterTypeArray.length > 0) {
+            this.queryParams[filterType] = filterTypeArray.join("~");
+        } else {
+            delete this.queryParams[filterType];
+        }
+        this.updateUrl(this.queryParams);
 
-            this.queryParams.q = this._convertQueryParams(this._chipList.filter(chip => chip.chipType === ChipType.CUSTOM_TEXT).map(chip => chip.chipElement), false);
-            this.updateUrl(this.queryParams);
+        if(filterType == FilterType.CUSTOM_FILTER_TIMINGS || filterType == FilterType.CUSTOM_FILTER_FAILURE_ASPECTS){
+            // since the Custom Filter Chip only appears when navigating from another view to the tests view,
+            // we can use it for failureAspects and Methods,because they never appear together
+            delete this.queryParams.methods;
+            delete this.queryParams.failureAspect;
         }
-        this._filter()
-    }
 
-    private _convertQueryParams(params, toArray: boolean){
-        // use "~" as separator because it is not used in id or class names and is not reserved in urls
-        if(toArray){
-            return params.split("~");
-        } else {
-            if(params == ""){
-                return;
-            }
-            return params.join("~");
+        // insert classes that are not selected as filter anymore back into filter class select options
+        if(filterType == FilterType.CLASS && !this._filteredClassStatistics.some(stat => stat.classIdentifier == filterObject)){
+            this._filteredClassStatistics.push(this._executionStatistics.classStatistics.find(stat => stat.classIdentifier == filterObject));
+            this._filteredClassStatistics.sort((a,b) => a.classIdentifier.localeCompare(b.classIdentifier));
         }
+
+        this.updateUrl(this.queryParams);
+
+        this._filter();
     }
 }
 
-interface IChip {
-    chipElement?: any;
-    chipType: ChipType;
+export enum FilterType {
+    CLASS="class",
+    CUSTOM_TEXT="q",
+    STATUS="status",
+    CUSTOM_FILTER_TIMINGS="methods",
+    CUSTOM_FILTER_FAILURE_ASPECTS="failureAspect"
+}
+
+type Filter = {
+    type: FilterType,
+    cssClass: string,
```

Comment:
Could have "name" property as well.

Reply:
See comment for ChipNameValueConverter
=> that is why i removed name property, because this way it would not be used anywhere